### ISSUE TITLE
k_hop_subgraph cannot maintain that the actual mini-batch nodes always be placed first

### ISSUE THREAD

**Issue by LukeLIN-web** [pyg-team/pytorch_geometric#5624]
### 🛠 Proposed Refactor

I am glad to find that actual mini-batch nodes always be placed first
https://github.com/pyg-team/pytorch_geometric/discussions/5279#discussioncomment-3814170 
But 
https://github.com/pyg-team/pytorch_geometric/blob/2d77047811311544d4861fa9e80fd2ec60ab6d01/torch_geometric/utils/subgraph.py#L302
`unique` always always sort the tensor [link](https://pytorch.org/docs/stable/generated/torch.unique.html) 
So that it cannot maintain that the actual mini-batch nodes always be placed first.

### Suggest a potential alternative/fix

I just use set.
```python
    origin_nodes = set([node_idx])
    for i in range(1,num_hops):
        subsets[i] = subsets[i][~np.isin(subsets[i],origin_nodes)]
```
But I don't know how to return_inverse correctly.

**Comment by rusty1s** [comment]
Yes, this holds only true for the `NeighborLoader`. I guess we could make use of the `neighbor_sample` routine in `NeighborLoader` to also guarantee that for `k_hop_subgraph`.

**Comment by LukeLIN-web** [comment]
> Yes, this holds only true for the `NeighborLoader`. I guess we could make use of the `neighbor_sample` routine in `NeighborLoader` to also guarantee that for `k_hop_subgraph`.

Thank you for your reply. But which routine are you specifying please? `torch.ops.torch_sparse.neighbor_sample`?

**Comment by rusty1s** [comment]
At best, we use the implementation from `pyg-lib` as used in the `NeighborLoader`, see [here](https://github.com/pyg-team/pytorch_geometric/blob/master/torch_geometric/sampler/neighbor_sampler.py#L282-L295). It is documented [here](https://pyg-lib.readthedocs.io/en/latest/modules/sampler.html).

**Comment by LukeLIN-web** [comment]
> At best, we use the implementation from `pyg-lib` as used in the `NeighborLoader`, see [here](https://github.com/pyg-team/pytorch_geometric/blob/master/torch_geometric/sampler/neighbor_sampler.py#L282-L295). It is documented [here](https://pyg-lib.readthedocs.io/en/latest/modules/sampler.html).

I found the doc, it seems that it have not been implemented?

**Comment by rusty1s** [comment]
Can you clarify what you mean?

**Comment by LukeLIN-web** [comment]
> Can you clarify what you mean?

I am sorry. I means I cannot found the code implement of torch.ops.pyg.neighbor_sample

**Comment by EdisonLeeeee** [comment]
Just wondering why we should support this as the mini-batch nodes are the input nodes specified as argument `node_idx`. Let me know if I missed something.

**Comment by rusty1s** [comment]
@LukeLIN-web The implementation is [here](https://github.com/pyg-team/pyg-lib/blob/master/pyg_lib/csrc/sampler/cpu/neighbor_kernel.cpp).
